lib/src/json_schema/validator.dart

```diff
@@ -106,17 +127,13 @@ class Validator {
   /// given context.
   List<int> _evaluatedItemsContext = [];
 
-  /// Validate the [instance] against the this validator's schema
-  bool validate(dynamic instance,
-      {bool reportMultipleErrors = false,
-      bool parseJson = false,
-      bool validateFormats,
-      bool treatWarningsAsErrors = false}) {
+  /// Validate the [instance] against the this validator's schema, returning results
+  ValidationResults validateWithResults(dynamic instance,
+      {bool reportMultipleErrors = false, bool parseJson = false, bool validateFormats}) {
```

Comment:
In most use-cases we've come across `reportMultipleErrors = true` is the default, as it is with `validateWithErrors`. I think we should continue that.